pkg/httpfs/prefixed.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2021 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpfs
+
+import (
+	"net/http"
+)
+
+// AddPrefixFS returns a FileSystem that adds a path prefix to all files
+// before delegating to the underlying fs.
+func AddPrefixFS(prefix string, fs http.FileSystem) http.FileSystem {
```

Comment:
nit: `AddPrefixFS` implies the func adds a prefix to the given filesystem, when in fact, it's creating a new struct. As such, I suggest renaming to `NewPrefixedFS`.

Reply:
Perhaps, on the other hand, std lib often uses naming pattern without New. And the business function is exactly what you said - adding prefix to a file system, returning a struct is besides the point. I think New is typically used when a public struct is returned, not in case of decorators.

Having said that, the FS prefix is rather unnecessary: httpfx.AddPrefix()